internal/deviations/deviations.go

```diff
@@ -99,4 +110,21 @@ var (
 	StaticProtocolName = flag.String("deviation_static_protocol_name", "DEFAULT", "The name used for the static routing protocol.  The default name in OpenConfig is \"DEFAULT\" but some devices use other names.")
 
 	DeprecatedVlanID = flag.Bool("deviation_deprecated_vlan_id", false, "Device requires using the deprecated openconfig-vlan:vlan/config/vlan-id or openconfig-vlan:vlan/state/vlan-id leaves.")
+
+	AddSubIntfToNetInst = flag.Bool("deviation_adding_subintf_to_network_instance", true,
```

Comment:
I don't think this should be a deviation.

If we agree that all tests must put subinterfaces into a VRF, someone should modify the tests to do that unconditionally.

By the way, my preferred name for the deviation would have been `--deviation_explicit_subinterface_in_vrf`.

Reply:
If the decision is to add each subinterface to a specific VRF explicitly, then that works, I will remove this deviation. The initial issue we had was that there was an assumption where the (sub)interfaces would be in the `DEFAULT` VRF.

Reply:
I think because we don't have this in all tests today, I'd like to go down the route of adding this to the deviations, and then subsequently the way that we can remove it is to determine whether:

a) the implementation here changes to take into account the difference, _or_
b) we explicitly specify the network instance for all interfaces in all deployments.

What I can do is write a specific test that covers interface to network instance assignment that we can use to determine whether we need this deviation in the medium term. I'll work on that now, but I would prefer that we move forward with this deviation for the moment.

cc: @xw-g @vcheny @vosipchu @sachendras for comment too.

Reply:
+1 to a deviation. Interface assignment to vrfs differs for vendors and making explicit assignment mandatory (specially to default vrf) will make some devices to fail.

Reply:
I think that there are two things here:

1. Is the interface implicitly assigned to a default NI?
2. Can one configure explicitly that an interface is part of the default NI?

It seems that both of these options can coexist.

Reply:
I checked in #758 to validate this -- needs to be run against implementations.

Reply:
After we validate #758 against other vendors, do we still need this as a deviation though?

Reply:
@robshakir, few things with #758 - I just ran a quick test.
1. The interface ID is not populated `i.Id = ygot.String(dp.Name() + "." + fmt.Sprint(p.Subintf))` which results in only the interface attempted to be assigned to the NI 
2. The update method exposes the issue that I was referring to [here](https://github.com/openconfig/featureprofiles/pull/737#discussion_r1016937740) when attempting to move an interface from one NI to another, it requires a delete (maybe Replace)?
```
[FailedPrecondition] Error in path: .network-instance{.name=="BLUE"}.interface{.name=="ethernet-1/19.0"}
            [FailedPrecondition] subinterface is already bound to .network-instance{.name=="DEFAULT"}.interface{.name=="ethernet-1/19.0"}
```

Reply:
I agree both options can coexist. However, If a device supports option 1, then the second option will be a no-op configuration. I think a device should not be tagged as no-compliant for not supporting a no-op configuration.

Also as @Hobbydos mentioned, there is another point that need to be addressed in the [test](https://github.com/openconfig/featureprofiles/pull/758) that is related to moving an active interface from one vrf to another that may not allowed by all devices.

Reply:
Basically we are deciding between:
1. Doing something that's allowed by current OC spec, but possible brings extra requirements to other 3 vendors and our current NPIs.
2. Set a deviation for only one vendor (and take time and effort to decide on the long term solution later, e.g. clarify the OC spec and ask the vendor OS to adjust the behavior), which will not introduce extra requirements to neither the NPIs nor other vendors.

In this case, if 1) is a no-op to all vendors OS, I would go with it. However, we know for sure that 1) is not a no-op action (and it's even a challenge) to at least 2 vendors.

Therefore I think we should go with 2), which is to introduce a deviation. 

cc: @vcheny @vosipchu @sachendras

Reply:
> @robshakir, few things with #758 - I just ran a quick test.
> 
> 1. The interface ID is not populated `i.Id = ygot.String(dp.Name() + "." + fmt.Sprint(p.Subintf))` which results in only the interface attempted to be assigned to the NI

On this issue -- the `interface` and `subinterface` leaves are what specify what interface + subinterface should be included, rather than the name in the key. If this specific name is required, this sounds like something we should file a bug against.

> 2. The update method exposes the issue that I was referring to [here](https://github.com/openconfig/featureprofiles/pull/737#discussion_r1016937740) when attempting to move an interface from one NI to another, it requires a delete (maybe Replace)?
> 
> ```
> [FailedPrecondition] Error in path: .network-instance{.name=="BLUE"}.interface{.name=="ethernet-1/19.0"}
>             [FailedPrecondition] subinterface is already bound to .network-instance{.name=="DEFAULT"}.interface{.name=="ethernet-1/19.0"}
> ```

Thanks - yes, I'd missed this writing the initial test. I'll update it.

Reply:
+1 on 2). My interpretation from  DEFAULT_INSTANCE network_instance_type below is that all interfaces should be in default vrf if not specified. 

A special routing instance which acts as the 'default' or 'global' routing instance for a network device.

Reply:
I don't want to be a pest, so if we want the tests to pass on a device that don't support VRF, they have a way to do it.